### PR TITLE
feat(core): introduce a new core package

### LINKED ISSUE #29: Extract classes to detect the type of specification into the core package
To reduce the number of duplicates and provide a dedicated way to detect the type of specification we should extract several classes into the new core package.

e.g.

```ts
enum ImporterType {
  HAR = 'har',
  OASV2 = 'oasv2',
  OASV3 = 'oasv3',
  POSTMAN = 'postman'
}

interface Spec<T> {
  readonly type: ImporterType;
  readonly doc: T;
  readonly name?: string;
}

interface ImportOptions {
  readonly mimeType: string;
}

interface Importer {
  type: ImporterType;

  importSpec(
    content: string,
    options?: ImportOptions
  ): Promise<ImportedSpec | undefined>;
}
```

### PR DESCRIPTION
**Pull request by derevnjuk**
closes #29

### CODE REVIEW

File: packages/validator/package.json [hunk 1]
```diff
@@ -38,10 +38,9 @@
   "keywords": [
     "swagger",
     "validator",
-    "yml",
-    "yaml",
-    "har",
-    "json"
+    "oas",
```
Comment (pmstss) [inline packages/validator/package.json:41]: i would add `openapi` as keyword as well

File: packages/core/src/importers/OASV3Importer.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { ImporterType } from './ImporterType';
+import { OpenAPIV3 } from '../types';
+import { BaseOASImporter } from './BaseOASImporter';
+
+export class OASV3Importer extends BaseOASImporter<ImporterType.OASV3> {
+  private readonly SUPPORTED_OPENAPI_VERSION = /^3\.\d+\.\d+$/; // 3.x.x
```
Comment (pmstss) [inline packages/core/src/importers/OASV3Importer.ts:6]: `validator` package supports only 3.0.x :/

File: packages/core/tests/OASV2Importer.spec.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        swagger: '2.0',
+        info: { version: 'v1', title: 'example' },
+        paths: {}
+      });
+
+      // assert
+      result.should.be.true;
+    });
+
+    it('should refuse to support of an incompatibility version', () => {
```
Comment (pmstss) [inline packages/core/tests/OASV2Importer.spec.ts:25]:
```suggestion
    it('should refuse to support of an incompatible version', () => {
```

File: packages/core/tests/HARImporter.spec.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+describe('HARImporter', () => {
+  const importer = new HARImporter();
+
+  describe('type', () => {
+    it(`should be equal ${ImporterType.HAR}`, () =>
+      importer.type.should.eq(ImporterType.HAR));
+  });
+
+  describe('isSupported', () => {
+    it('should refuse to support of an incompatibility version', () => {
```
Comment (pmstss) [inline packages/core/tests/HARImporter.spec.ts:13]:
```suggestion
    it('should refuse to support of an incompatible version', () => {
```

File: packages/core/tests/OASV3Importer.spec.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+describe('OASV3Importer', () => {
+  const importer = new OASV3Importer();
+
+  describe('type', () => {
+    it(`should be equal ${ImporterType.OASV3}`, () =>
+      importer.type.should.eq(ImporterType.OASV3));
+  });
+
+  describe('isSupported', () => {
+    it('should refuse to support of an incompatibility version', () => {
```
Comment (pmstss) [inline packages/core/tests/OASV3Importer.spec.ts:13]:
```suggestion
    it('should refuse to support of an incompatible version', () => {
```

File: packages/core/tests/PostmanImporter.spec.ts [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        },
+        item: [],
+        variable: []
+      });
+
+      // assert
+      result.should.be.true;
+    });
+
+    it('should refuse to support of an incompatibility version', () => {
```
Comment (pmstss) [inline packages/core/tests/PostmanImporter.spec.ts:43]:
```suggestion
    it('should refuse to support of an incompatible version', () => {
```

File: packages/core/tests/first.spec.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
+describe('first', () => {
+  it('should be resolved with an undefined if all promises are rejected');
```
Comment (pmstss) [inline packages/core/tests/first.spec.ts:2]: missing test cases code?

File: packages/core/README.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+npm i --save @har-sdk/core
+```
+
+## Usage
+
+```ts
+import { SpecExplorer } from '@har-sdk/core';
+import petstore from './petstore.collection.json';
+
+const result = await new SpecExplorer().tryToImportSpec(petstore);
```
Comment (pmstss) [inline packages/core/README.md:17]: i'm not a fan of current naming (both for class and method), but have no better ideas atm

**Review by pmstss — CHANGES_REQUESTED** [review]
noop first.spec.ts

File: packages/core/README.md (hunk 8 above)
Comment (derevnjuk) [reply packages/core/README.md:17]: Yeah, me too. We can discuss this point

File: packages/core/tests/first.spec.ts (hunk 7 above)
Comment (derevnjuk) [reply packages/core/tests/first.spec.ts:2]: Nope, this is a pending test (aka todo). Use them to define tests cases, later you can back and implement them, knowing that you don't forget anything

**Review by pmstss — APPROVED** [review]

**Pull request merged**